packages/core/addon/components/field-renderer.js

```diff
@@ -23,6 +39,7 @@ export default class FieldRenderer extends Component {
     if (this.args.field.nonce !== this.currentNonce) {
       this.currentNonce = this.args.field.nonce;
       this.newFieldName = this.args.field.name;
+      this.newFieldLabel = this.args.field.label;
```

Comment:
@robustdj this is the key line right here that makes the label's `<Input>` one way bound. this is how the upstream state gets set back into the input component.